service/paging.go

```diff
@@ -0,0 +1,64 @@
+package service
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+// PagingParams represents paging and sorting parameter values
+type PagingParams struct {
+	PageSize   *int32
+	PageNumber *int32
+	SortField  string
+	SortAsc    bool
+}
+
+// ParsePagingParams retrieves paging params from the request, allows for whitelisting sort field values
+func ParsePagingParams(r *http.Request, sortFieldsWhitelist []string) (PagingParams, error) {
+	paging := PagingParams{}
+
+	// page number
+	pageNumber, err := Int32PointerFromQueryParam(r, "page_number")
+	if err != nil {
+		return paging, err
+	}
+	paging.PageNumber = pageNumber
+
+	// page size
+	pageSize, err := Int32PointerFromQueryParam(r, "page_size")
+	if err != nil {
+		return paging, err
+	}
+	paging.PageSize = pageSize
+
+	// sort asc
+	sortAscStr := r.URL.Query().Get("sort_asc")
+	if len(sortAscStr) > 0 {
+		sortAsc, err := strconv.ParseBool(sortAscStr)
+		if err != nil {
+			return paging, err
+		}
+		paging.SortAsc = sortAsc
+	}
+
+	// sort field
+	sortField := r.URL.Query().Get("sort_field")
+	if !contains(sortFieldsWhitelist, sortField) {
```

Comment:
What if sortField is not included? There isn't a test.  It looks like it would pass nil as a string argument, which would fail.

Reply:
good call!  as a rule, i think we should always have default values for all of the sort params in the stored proc.  currently i'm checking for nil on those.  right now the parser errors on missing sort field (basically due to "" not being whitelisted).  i'm going to change that to not error and rely on the default.